python/nvtx/_lib/lib.pyx

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+from nvtx._lib.lib cimport *
+from nvtx.colors import color_to_hex
+from nvtx.utils.cached import CachedInstanceMeta
+
+cdef extern from "Python.h":
+    wchar_t* PyUnicode_AsWideCharString(object, Py_ssize_t *)
+
+
+def initialize():
+    nvtxInitialize(NULL)
+
+
+cdef class EventAttributes:
+    cdef unicode _message
+    cdef object _color
+    cdef nvtxEventAttributes_t c_obj
+
+    def __init__(self, unicode message=None, color=None):
+        if message is None:
+            message = ""
+        self._message = message
+        self._color = color
+        self.c_obj = nvtxEventAttributes_t(0)
+        self.c_obj.version = NVTX_VERSION
+        self.c_obj.size = NVTX_EVENT_ATTRIB_STRUCT_SIZE
+        self.c_obj.colorType = NVTX_COLOR_ARGB
+        self.c_obj.color = color_to_hex(self._color)
+        self.c_obj.messageType = NVTX_MESSAGE_TYPE_UNICODE
+        self.c_obj.message.unicode = PyUnicode_AsWideCharString(
+            self._message,
+            NULL
+        )
```

Comment:
May want to add a comment that a future TODO is to expose a fastpath for ASCII encoding